### PR TITLE
Return Errors for failed transactions

### PR DESCRIPTION
**Pull request by evgenykuzyakov**
Depends on nearlib update to handle it.

### CODE REVIEW

File: test-utils/testlib/src/lib.rs [hunk 1]
```diff
@@ -11,6 +11,9 @@ use near_store::test_utils::create_test_store;
 use std::sync::Arc;
 use tempdir::TempDir;
 
+#[macro_use]
+extern crate assert_matches;
```
Comment (frol) [inline test-utils/testlib/src/lib.rs]: Rust 2018 edition deprecated `extern crate` with simple `use` statements (you just import the macro in the module where you used it like this: `use assert_matches::assert_matches;`)

File: runtime/runtime/tests/test_evil_contracts.rs [hunk 2]
```diff
@@ -4,6 +4,10 @@ use near_primitives::views::FinalExecutionStatus;
 use std::mem::size_of;
 use testlib::node::{Node, RuntimeNode};
 
+#[cfg(test)]
+#[macro_use]
+extern crate assert_matches;
```
Comment (frol) [inline runtime/runtime/tests/test_evil_contracts.rs]: `extern crate` ^^^

**Comment by ilblackdragon** [comment]
@evgenykuzyakov (also CC @SkidanovAlex ) in case of challenges - we actually might have invalid tx in the block.
For example:
 - tx that moves money from account of the challenge originator is included in chunk that is included in block t
 - malicious challenge is included in block header t
 - it all gets executed in t+1, where challenge takes presendence and slashes all the money making the subsequent tx invalid.

**Comment by evgenykuzyakov** [comment]
Nearlib change for this PR is pending: https://github.com/nearprotocol/nearlib/pull/85

**Comment by evgenykuzyakov** [comment]
@ilblackdragon Not sure why invalid transactions are relevant to this error handling

**Comment by evgenykuzyakov** [comment]
> @ilblackdragon Not sure why invalid transactions are relevant to this error handling
Talked to @SkidanovAlex and now I understand what you mean. I suggest we pass a list of accounts who were successfully challenged to the Runtime. So signed_transactions from these account IDs could be ignored without rejecting the entire chunk.

**Pull request merged**